<launch>
	<node pkg="tf" type="static_transform_publisher" name="world_odom_broadcaster" args="0 0 0 0 1 0 0 /world /odom 100" />
	<group ns="position_controller">
		<node pkg="position_controller" type="position_controller_node" 
			name="position_controller_node" output="screen" 			
			><!--launch-prefix="gdbserver localhost:10000"-->
			<param name="motor_controller_output_topic" type="string" value="/robot/motion"/>  
			<param name="motor_controller_input_topic" type="string" value="/robot/input_control"/> 
			<param name="odometry_output_topic" type="string" value="/position_controller/pose"/> 
			<param name="height_range_input_topic" type="string" value="/robot/sonar"/> 
            <param name="flow_input_topic" type="string" value="/robot/flow"/>
            <param name="flow_rot_input_topic" type="string" value="/robot/flow_rad"/>  
            <param name="input_pose_topic" type="string" value="/position_controller/target_pose"/>
			<param name="altitude_max_incline_error" value="1"/>
			<param name="altitude_max_decline_error" value="-1"/> 	
			<param name="thrust_hover" value="125"/>
			<param name="thrust_max_change" value="2" />
			<param name="yaw_max_velocity" value="2" />	
			<param name="control_pitch_roll_factor" value="0.005" />
			<param name="control_yaw_factor" value="0.005" />			
			<param name="max_velocity" value="2" />		
			<param name="max_pitch_roll" value="50" />	
			<param name="max_yaw" value="127" />	
			<param name="px4flow_quality_threshold" value="150" />	
			<param name="external_distance_sensor" value="true" />	
			<param name="simulation_mode" value="true" />         				
		</node>
		<group ns="altitude_pid" >
			<param name="p" value="1" />
			<param name="i" value="0.05" />
			<param name="d" value="3" />
			<param name="i_clamp" value="20" />
			<param name="antiwindup" value="true" />			
		</group>
		<group ns="pitch_pos_pid" >
			<param name="p" value="0.5" />
			<param name="i" value="0.02" />
			<param name="d" value="2" />
			<param name="i_clamp" value="10" />
			<param name="antiwindup" value="true" />	
		</group>
		<group ns="roll_pos_pid" >
			<param name="p" value="0.5" />
			<param name="i" value="0.02" />
			<param name="d" value="2" />
			<param name="i_clamp" value="10" />
			<param name="antiwindup" value="true" />	
		</group>
		<group ns="pitch_pid" >
			<param name="p" value="50" />
			<param name="i" value="0" />
			<param name="d" value="2" />
			<param name="i_clamp" value="10" />
			<param name="antiwindup" value="true" />	
		</group>
		<group ns="roll_pid" >
			<param name="p" value="50" />
			<param name="i" value="0" />
			<param name="d" value="2" />
			<param name="i_clamp" value="10" />
			<param name="antiwindup" value="true" />	
		</group>
		<group ns="yaw_pid" >
			<param name="p" value="10" />
			<param name="i" value="2.5" />
			<param name="d" value="0" />
			<param name="i_clamp" value="10" />
			<param name="antiwindup" value="true" />	
		</group>
		<group ns="yaw_angle_pid" >
			<param name="p" value="3" />
			<param name="i" value="0.01" />
			<param name="d" value="0" />
			<param name="i_clamp" value="2" />
			<param name="antiwindup" value="true" />	
		</group>
	</group>
</launch>

